package/MDAnalysis/auxiliary/base.py

```diff
@@ -678,8 +786,23 @@ def calc_representative(self):
             except KeyError:
                 value = cutoff_data[min_diff]
         elif self.represent_ts_as == 'average':
-            value = np.mean(np.array([val for val in cutoff_data.values()]),
-                            axis=0)
+            try:
+                value = np.mean(np.array(
+                                [val for val in cutoff_data.values()]
+                                ), axis=0)
+            except TypeError:
+                # for readers like EDRReader, the above does not work
+                # because each step contains a dictionary of numpy arrays
+                # as data
+                value = {}
+                for dataset in cutoff_data:
+                    for term in self.terms:
+                        if term not in value:
+                            value[term] = cutoff_data[dataset][term]
+                        else:
+                            value[term] += cutoff_data[dataset][term]
```

Comment:
Using a [`defaultdict`](https://docs.python.org/3.8/library/collections.html#collections.defaultdict) avoids the if/else. Or initialize a dict with all the terms and zeros. But avoid branches insides loops if you can.

(It might not be a big performance penalty here but I think it's always good to strive for cleaner code.)